lib/bats-core/test_functions.bash

```diff
@@ -80,7 +80,7 @@ skip() {
 bats_test_begin() {
   BATS_TEST_DESCRIPTION="$1"
   if [[ -n "$BATS_EXTENDED_SYNTAX" ]]; then
-    printf 'begin %d %s\n' "$BATS_TEST_NUMBER" "$BATS_TEST_DESCRIPTION" >&3
```

Comment:
Yeah this (among other examples of `BATS_TEST_NUMBER`) is what leads me to believe that the original behaviour of `BATS_TEST_NUMBER` was actually a bug that got documented as a feature.

Reply:
Yes, I am with you on that. (Probably because I tried to preserve your implementation's behavior in my JUnit/setup_file rewrite.) However, due to the lack of setup_file I fear there are many codebases that rely on the old behavoir.